common/types/activation.go

```diff
@@ -173,6 +174,45 @@ func (atx *ActivationTx) InnerBytes() ([]byte, error) {
 	return InterfaceToBytes(atx.InnerActivationTx)
 }
 
+// Fields returns an array of LoggableFields for logging
+func (atx *ActivationTx) Fields(layersPerEpoch uint16, size int) []log.LoggableField {
+	commitmentStr := "nil"
```

Comment:
I think I made it log "nil" rather than an empty string because it's more readable to human eyes. I'd prefer to err on the side of keeping the code simpler, especially for logs. Let me know if you disagree.

Reply:
I agree that the string "nil" is better than an empty string and I understand your consideration of keeping the code simple. Having said that, I don't think this would complicate the code... You already have a conditional that overwrites the "nil" string.

I don't want to impose my style onto you, so feel free to reject what I suggest if you disagree, but what I had in mind is writing the function like this:
```golang
func (atx *ActivationTx) Fields(layersPerEpoch uint16, size int) []log.LoggableField {
	fields := []log.LoggableField{
		atx.ID(),
		log.String("sender_id", atx.NodeID.ShortString()),
		log.String("prev_atx_id", atx.PrevATXID.ShortString()),
		log.String("pos_atx_id", atx.PositioningATX.ShortString()),
		atx.PubLayerID,
		atx.PubLayerID.GetEpoch(layersPerEpoch),
		log.Uint32("active_set", atx.ActiveSetSize),
		log.Int("view_len", len(atx.View)),
		log.Uint64("sequence_number", atx.Sequence),
		log.Int("atx_size", size),
	}
	if atx.Commitment != nil {
		fields = append(fields, log.String("commitment", atx.Commitment.String()))
	}
	if h, err := atx.NIPSTChallenge.Hash(); err == nil && h != nil {
		fields = append(fields, log.String("nipst_challenge", h.String()))
	}
	return fields
}
```

Reply:
I changed "nil" to the empty string here, so that `commitmentStr` and `challenge` are handled the same way. However, I disagree with leaving the fields out entirely if they're empty. Having spent countless hours of my life looking at logs and hunting down bugs, I'd definitely prefer to err on the side of being explicit so that it's immediately obvious that the fields are present in the data structure, and are empty in this instance, but may not always be empty.

Thanks for your feedback here!